tao_compiler/mlir/disc/transforms/disc_gpu_kernel_to_blob.cc

```diff
@@ -417,6 +417,53 @@ class GpuKernelToBlobPass
       TF_ASSIGN_OR_RETURN(gpu_asm,
                           tensorflow::se::BundleGpuAsm({image}, gpu_asm_opts));
     }
+
+    auto* env = tsl::Env::Default();
+    std::vector<std::string> tempdir_vector;
+    env->GetLocalTempDirectories(&tempdir_vector);
+    if (tempdir_vector.empty()) {
+      return xla::InternalError(
+          "Unable to locate a temporary directory for compile-time artifacts.");
+    }
+    std::string tempdir_name = tempdir_vector.front();
+    VLOG(1) << "Compile-time artifacts located at: " << tempdir_name;
+    std::string random_number = std::to_string(tensorflow::random::New64());
+    std::string filename =
+      absl::StrCat(llvmModule->getModuleIdentifier(), random_number);
+    std::string ir_path =
+      tensorflow::io::JoinPath(tempdir_name, absl::StrCat(filename, ".ll"));
+    std::string ptx_path =
+      tensorflow::io::JoinPath(tempdir_name, absl::StrCat(filename, ".ptx"));
+    std::string cubin_path =
+      tensorflow::io::JoinPath(tempdir_name, absl::StrCat(filename, ".cubin"));
+
+    // Dump LLVM IR.
+    std::error_code ec;
+    std::unique_ptr<llvm::raw_fd_ostream> ir_fs(
+        new llvm::raw_fd_ostream(ir_path, ec, llvm::sys::fs::OF_None));
+    llvmModule->print(*ir_fs, nullptr);
+    ir_fs->flush();
+
+    // Dump PTX.
+    auto status = tsl::WriteStringToFile(env, ptx_path, ptx);
+    if (!status.ok()) {
+      LOG(ERROR) << "Could not write PTX to " << ptx_path << ": " << status;
+    }
+
+    // Dump cubin.
+    std::ofstream fout(cubin_path, std::ios::binary);
+    fout.write(reinterpret_cast<const char*>(gpu_asm.data()), gpu_asm.size() * sizeof(uint8_t));
+    fout.close();
+
+    bool keep_tempfiles = false;
+    TF_CHECK_OK(tensorflow::ReadBoolFromEnvVar("DISC_CUDA_KEEP_TEMPFILES",
+                                                /*default_val=*/false,
+                                                &keep_tempfiles));
+    if (!keep_tempfiles) {
```

Comment:
how about move `if` to L420 to guard all the code above? Because if one doesn't want to keep temp files, we even don't have to dump it.

Reply:
Done